firmware/ethereum.c

```diff
@@ -280,6 +280,31 @@ static void layoutEthereumConfirmTx(const uint8_t *to, const uint8_t *value, uin
  * - data (0 ..)
  */
 
+static bool ethereum_signing_check(EthereumSignTx *msg)
+{
+	// determine if address == 0
+	bool address_zero = msg->has_to;
+	if (address_zero) {
+		for (size_t i = 0; i < msg->to.size; i++) {
+			if (msg->to.bytes[i] > 0) {
+				address_zero = false;
+				break;
+			}
+		}
+	}
+
+	// sending value to address 0
+	if (address_zero && msg->has_value && msg->value.size) {
```

Comment:
Sorry I've realised that sending value is valid, but only part of contract deployment (initial stipend).

Basically the following check alone should be enough.

We could also check for sending data to a non-zero address which is most likely a contract invocation.We could warn the user it is not a regular transfer.